### PR TITLE
Update solidity docs

### PR DESCRIPTION
**Pull request by tremblaythibaultl**
- Introduce the CMUX operator
- Remove encrypted requires

### CODE REVIEW

File: docs/solidity/library.md [hunk 1]
```diff
@@ -50,5 +50,6 @@ The list of supported operations is presented below.
 | Max                   | `TFHE.max`    | Binary |
 | Neg                   | `TFHE.neg`    | Unary  |
 | Not                   | `TFHE.not`    | Unary  |
+| Cmyx                  | `TFHE.cmux`   | Ternary|
```
Comment (immortal-tofu) [inline docs/solidity/library.md:53]: Cmyx ?

File: docs/solidity/functions.md [hunk 2]
```diff
@@ -146,6 +146,203 @@ function gt(uint32 a, euint16 b) internal view returns (ebool)
...
+
+### Example
+```solidity
+// if (b == true) return val1 else return val2
+function cmux(ebool b, euint8 val1, euint8 val2) internal view returns (euint8) {
+    return TFHE.cmux(b, val1, val2);
+}
+```
+
+## `min`, `max`
```
Comment (immortal-tofu) [inline docs/solidity/functions.md:319]: min max seems to exists below? Error in copy/paste?

File: docs/solidity/functions.md [hunk 3]
```diff
@@ -146,6 +146,203 @@ function gt(uint32 a, euint16 b) internal view returns (ebool)
...
+
+```solidity
+// min(a, b)
+function min(euint32 a, euint16 b) internal view returns (euint32)
+
+// max(a, b)
+function max(uint32 a, euint8 b) internal view returns (euint32)
+```
+
+## Unary operators (`neg`, `not`)
```
Comment (immortal-tofu) [inline docs/solidity/functions.md:333]: Same, it seems you copy paste everything twice. I don't understand

**Review by immortal-tofu — CHANGES_REQUESTED** [review]

File: docs/solidity/library.md (hunk 1 above)
Comment (tremblaythibaultl) [reply docs/solidity/library.md:53]: Typo, fixed.

File: docs/solidity/decryption.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
+# Decryption and control structures
+
+## Decryptions
+
+We allow explicit decryption requests for any encrypted type.
+The values are decrypted through the distributed decryption protocol and are stored on-chain.
+
+### Example
```
Comment (immortal-tofu) [inline docs/solidity/decryption.md:8]: Maybe add example with require() ?

Comment (immortal-tofu) [reply docs/solidity/decryption.md:8]: I mean another example (one with `return` and one with `require()`)

Comment (tremblaythibaultl) [reply docs/solidity/decryption.md]: Good idea, done ✅

**Pull request merged**